docs/environment-variables.md

```diff
@@ -0,0 +1,170 @@
+
+### Common for all XL Products docker images,
+
+##### `APP_ROOT   ????`
+- value: "/opt/xebialabs"
+- Note: 
+    - not valid anymore as APP_ROOT is set to /opt/xebialabs .. however in db-drivers.sh if you set APP_ROOT to different path it will take affect but then scrib will be broken and will fail with below,
+cp: cannot stat '/opt//db-libs/h2*': No such file or directory
+and this will cause docker container to stop as DB libs can not be found.
+
+
+##### `APP_HOME  ????`
+- value: "/${APP_ROOT}/xl-deploy-server"
+- Note: 
+    - not valid anymore as APP_HOME is set to ${APP_ROOT}/{{ product }}-server .. if we tried to pass it as env docker container start will fail with below,
+/opt/xebialabs/xl-deploy-server/bin/run-in-container.sh: line 136: cd: /opt/default-plugins: No such file or directory
+
+##### `ADMIN_PASSWORD`
+- value: "password-value"
+- Admin user password used in XL Products
+- Note: 
+    - If this value is not passed a random 8 characters password will be generated using "pwgen 8"
+
+##### `REPOSITORY_KEYSTORE_PASSPHRASE`
+value: "passphrase-string" 
+Repository keystore passphrase used in XL Products
+If this value is not passed a random 16 characters password will be generated using "pwgen 8"
+Note: 
+    - Keystore password must be at least 6 characters
+
+##### `REPOSITORY_KEYSTORE`
+- value: "keystore-string"
+- Repository keystore passphrase used in XL Products
+- Note:    
+    - If this value is set then it will be converted to base64 value and added to keystore file like below,
+echo "${REPOSITORY_KEYSTORE}" | base64 -d > ${APP_HOME}/conf/repository-keystore.jceks
+    - If this value is not set then new one will be created using keystore passphrase like below,
+keytool -genseckey -alias deployit-passsword-key -keyalg aes -keysize 128 -keypass "deployit" -keystore ${APP_HOME}/conf/repository-keystore.jceks -storetype jceks -storepass ${REPOSITORY_KEYSTORE_PASSPHRASE}
+
+##### `XL_LICENSE`
+- value: "license-string"
+- License if you want to explicitly provide it .. which will be added to license file.
+other option is to add license file directly to below path 
+${APP_HOME}/conf/xl-release-license.lic for XLRelease or ${APP_HOME}/conf/deployit-license.lic for XLDeploy
+
+##### `XL_LICENSE_KIND   ????`
+- value: "byol"
+- Bring your owen license .. if set then no request for trial license will be issued.
+- Note: 
+    - Something wrong here as even if this is set you still need to accept ACCEPT_EULA .. however as far as i understand it should not ask for trial license.
+
+##### `XL_NO_UNREGISTERED_LICENSE  ????`
+- value: "yes"
+- If passed then this will request unregistered license in which a call will be done to XebiaLabs license server to get a license and add it to license file. 
+- Note: 
+    - something wrong here as even if this is set you still asked to enter trial license as far as i understand it should not be the case.
+
+##### `ACCEPT_EULA`
```

Comment:
this was used for jetpack wright?

Reply:
Not Only Jetpack .. it is also needed if you will not provide license while starting container otherwise it will fail to start with below error,
 "You must accept the End User License Agreement or provide your own license before this container can start."